### PR TITLE
feat: Handle custom cluster-domain values

### PR DESCRIPTION
**Pull request by lucchmielowski**
Hi 👋 

This PR aims at fixing a "bug" (couldn't find a related issue though) in clusters that don't use `cluster.local` as their internal domain such as:

```txt
Error while dialing dial tcp: lookup keda-operator.keda.svc.cluster.local on 169.254.20.10:53: no such host"
```

It adds a `clusterDomain` value that defaults to `cluster.local` but that can be changed when needed.

This PR also removes `.svc.local` names from the certificates as well as it doesn't seem to be a valid address for kube's internal DNS 🤔 (but I'm no expert on kube DNS so that might be a mistake 😅.)

### Checklist

- [x] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
- [x] README is updated with new configuration values *(if applicable)*
- [x] ~A PR is opened to update KEDA core ([repo](https://github.com/kedacore/keda)) *(if applicable, ie. when deployment manifests are modified)*~

### CODE REVIEW

**Comment by lucchmielowski** [comment]
I'm not 100% sure how to update the documentation TBH 😓 (should I just run `helm docs` ?)

**Comment by JorTurFer** [comment]
@zroubalik , could you take a look?

**Comment by lucchmielowski** [comment]
Thanks for the answer @JorTurFer ! 🙏. Made the change inside the README file.

File: keda/Chart.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ kubeVersion: ">=v1.23.0-0"
 
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
-version: 2.9.4
+version: 2.10.0
```
Comment (tomkerkhove) [inline keda/Chart.yaml:11]: Please revert this

Comment (lucchmielowski) [reply keda/Chart.yaml]: Done @tomkerkhove 👍

**Review by zroubalik — APPROVED** [review]
LGTM

**Comment by marandalucas** [comment]
@lucchmielowski Hi! Thank you so much for this fix. 

Unfortunately, It doesn't work for us.

- We noticed that we should install "cert-manager".
- metrics-service-address is hardcoded. So, we can't update it from values.yaml.
- Certificates and secure connections are mandatory.

HELM CONFIG
`clusterDomain: gcp-prod-pv-na1-a.company.cluster.local
`

ERROR:
`W0314 15:03:14.706154       1 logging.go:59] [core] [Channel #1 SubChannel #2] grpc: addrConn.createTransport failed to connect to {Addr: "keda-operator.keda.svc.gcp-prod-pv-na1-a.company.cluster.local:9666", ServerName: "keda-operator.keda.svc.gcp-prod-pv-na1-a.company.cluster.local:9666", }. Err: connection error: desc = "transport: authentication handshake failed: tls: failed to verify certificate: x509: certificate is valid for keda-operator, keda-operator, keda-operator.keda, keda-operator.keda.svc, keda-operator.keda.svc.cluster.local, keda-admission-webhooks, keda-admission-webhooks.keda, keda-admission-webhooks.keda.svc, keda-admission-webhooks.keda.svc.cluster.local, keda-operator-metrics-apiserver, keda-operator-metrics-apiserver.keda, keda-operator-metrics-apiserver.keda.svc, keda-operator-metrics-apiserver.keda.svc.cluster.local, not keda-operator.keda.svc.gcp-prod-pv-na1-a.company.cluster.local"`

We wonder if you could fix it. We don't need cert-manager in our clusters.

Thanks in advance

**Pull request merged**